modules/apps/notification/notification-service/src/main/java/com/liferay/notification/internal/type/NotificationTypeRegistryImpl.java

```diff
@@ -36,6 +38,12 @@ public class NotificationTypeRegistryImpl implements NotificationTypeRegistry {
 
 	@Override
 	public NotificationType getNotificationType(String key) {
+		if (!GetterUtil.getBoolean(PropsUtil.get("feature.flag.LPS-158482")) &&
```

Comment:
It's best to simply not register the service when the feature flag property is not true.
Also, the goal of this feature is to allow the creation of notification templates using system object definitions. Therefore, we should be blocking any attempt to include a system object definition into the column objectDefinitionId of the NotificationTemplate entity. Go to the local service of this entity and check line 566.